commands/cloudapi-v6/networkloadbalancer.go

```diff
@@ -280,21 +288,30 @@ func RunNetworkLoadBalancerUpdate(c *core.CommandConfig) error {
 }
 
 func RunNetworkLoadBalancerDelete(c *core.CommandConfig) error {
-	if err := utils.AskForConfirm(c.Stdin, c.Printer, "delete network load balancer"); err != nil {
-		return err
-	}
+	var resp *resources.Response
 	dcId := viper.GetString(core.GetFlagName(c.NS, cloudapiv6.ArgDataCenterId))
 	nlbId := viper.GetString(core.GetFlagName(c.NS, cloudapiv6.ArgNetworkLoadBalancerId))
-	c.Printer.Verbose("NetworkLoadBalancer with id: %v is deleting...", nlbId)
-	resp, err := c.CloudApiV6Services.NetworkLoadBalancers().Delete(dcId, nlbId)
-	if resp != nil {
-		c.Printer.Verbose(cloudapiv6.RequestTimeMessage, resp.RequestTime)
-	}
-	if err != nil {
-		return err
-	}
-	if err = utils.WaitForRequest(c, waiter.RequestInterrogator, printer.GetId(resp)); err != nil {
-		return err
+	allFlag := viper.GetBool(core.GetFlagName(c.NS, cloudapiv6.ArgAll))
+	if allFlag {
+		err := DeleteAllNetworkLoadBalancers(c)
+		if err != nil {
+			return err
+		}
+	} else {
+		if err := utils.AskForConfirm(c.Stdin, c.Printer, "delete network load balancer"); err != nil {
+			return err
+		}
+		c.Printer.Verbose("Starting deleting NetworkLoadBalancer with id: %v...", nlbId)
+		resp, err := c.CloudApiV6Services.NetworkLoadBalancers().Delete(dcId, nlbId)
```

Comment:
Use = instead of := to not define resp as a new variable. It will not be recognized at the end of the current function.